pkg/tools/base/default.go

```diff
@@ -9,40 +9,69 @@ import (
 	"github.com/openshift/backplane-tools/pkg/utils"
 )
 
+var InstallDir = func() string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(fmt.Errorf("failed to retrieve $HOME dir: %w", err))
+	}
+	return filepath.Join(homeDir, ".local", "bin", "backplane")
+}()
+
+var LatestDir = func() string {
+	return filepath.Join(InstallDir, "latest")
+}()
+
 type Default struct {
-	Tool
-	Name             string
-	executableName   string // if empty, name will be used
+	// Name defines the 'formal' name this tool is referred to within this program
+	name string
+
+	// executableName defines the invokable used to run the tool via CLI after it has
+	// been installed
+	executableName string
+
+	// installedVersion is the currently installed version of the tool
 	installedVersion string
-	latestVersion    string
+
+	// latestVersion is the latest version of the tool available for install
+	latestVersion string
+}
+
+// NewDefault creates a Default tool with the provided name
+func NewDefault(name string) Default {
+	d := Default{
+		name:           name,
+		executableName: name,
+	}
+	return d
+}
+
+// NewDefaultWithExecutable creates a Default tool with the provided name and executableName
+func NewDefaultWithExecutable(name, executable string) Default {
```

Comment:
I hope we will not have too much optional fields otherwise we will have to create a Newer for each cardinality, that was the reason I kept the constructor with the named fields, but for the moment I agree it looks better like that ;)

Reply:
Agreed - we should keep both the number of constructors small, as well as the number of arguments they accept :slightly_smiling_face: